src/Checkers/LicenseChecker.js

```diff
@@ -0,0 +1,303 @@
+        pos = this._skipToNextLine(text, pos);
+        continue;
+      }
+      if (!/\s/.test(char)) {
+        break;
+      }
+      pos++;
+    }
+    return pos;
+  }
+
+  /**
+   * Skips to the next line.
+   * @private
+   * @return {integer} position
+   */
+  _skipToNextLine(text, pos) {
+    while (pos < text.length && text[pos] != '\n') {
+      pos++;
+    }
+    if (pos != text.length) pos++;
+    return pos;
+  }
+
+  _createJsonResponse(token, lineNum, linePos, currentPos) {
+    return {
+      token: token,
+      lineNum: lineNum,
+      linePos: currentPos - linePos
+    };
+  }
+
+  /**
+   * Skips to the text after the single line comments ("//") starting from _pos.
+   *
+   * @private
+   * @return offset from the current position (_pos) to the commented text or null if no comments found.
+   */
+  _skipToTextInComments(text, startPos) {
+    if (startPos + 1 >= text.length) return null;
+    if (text[startPos] != '/' || text[startPos + 1] != '/') {
+      return null;
+    } else {
+      return startPos + 2;
+    }
+  }
+
+  /**
+   * Check path for License mistakes
+   * @param {String} path
+   * @return {[ErrorMessage]}
+   */
+  check(dirpath) {
+    const allFiles = this._getFiles(dirpath, []);
+    const errors = [];
+    errors.push(false); // reserved for LICENSE error
+    for (const file of allFiles) {
+      const parsedPath = path.parse(file);
+      if (this._extensionsSet.has(parsedPath.ext)) {
+        errors.push(this._compareWithLicense(file, true));
+      } else if (parsedPath.name === 'LICENSE') {
+        errors[0] = this._compareWithLicense(file, false); // assign to reserved place
+      }
+    }
+    return errors.filter((error) => error != false);
+  }
+
+  /**
+   * Check path for License mistakes
+   * @param {String} filepath path to file
+   * @param {Boolean} isCode mark, should we skip comments and headers
+   * @return {ErrorMessage}
+   */
+  _compareWithLicense(filepath, isCode) {
+    const checkedLicense = fs.readFileSync(filepath, 'utf-8').replace(/\d\d\d\d(\-\d\d\d\d)?/, '');
+
+    const originalLicense = fs.readFileSync(LICENSE_FILE_PATH, 'utf-8');
+
+    const output = this._compareTwoLicenseTexts(checkedLicense, originalLicense, isCode);
+    return output ? new ErrorMessage(output.message, output.lineNum, filepath, 'LicenseChecker', output.linePos) : output;
+  }
+
+  /**
+   * Compare two licenses for equality
+   *
+   * @param {String} checkedText text of license or code
+   * @param {String} originalText original license
+   * @param {Boolean} isCode mark, should we skip comments and headers in checkedText
+   * @return {false | MessageJson} false if licenses are equal, message with differs otherwise
+   */
+  _compareTwoLicenseTexts(checkedText, originalText, isCode) {
+    const testedGen = this._tokensIterator(checkedText, isCode);
+    const goldenGen = this._tokensIterator(originalText, false);
+
+    let checkedToken = testedGen.next().value;
```

Comment:
So it took everything you had to rename the two variables above and you can't make it for the following lines anymore? )